sprints/dashboard/libs/mattermost.py

```diff
@@ -0,0 +1,70 @@
+from contextlib import contextmanager
+from typing import Iterator
+from mattermostdriver import Driver
+from django.conf import settings
+
+
+class Mattermost:
+    """ Class that is responsible to connect to mattermost bot.
+    """
+
+    def __init__(self, username: str, access_token: str) -> None:
+        """
+        Args:
+            username: Mattermost username of the user.
+            access_token: Access token for the bot user.
+
+        """
+        connection_parameter = {
+            'url': settings.MATTERMOST_SERVER,
+            'port': settings.MATTERMOST_PORT,
+            'username': username,
+            'token': access_token
+        }
+        self.mattermost_connection = Driver(connection_parameter)
+        self.mattermost_connection.login()
+
+    def get_username_from_email(self, email_id: str) -> str:
+        """
+        Funtion that helps to get mattermost usrename
+        from email id.
+
+        Args:
+            email_id: Email id of the user.
+
+        Returns:
+            str: username.
+        """
+        username = self.mattermost_connection.users.get_user_by_email(email=email_id)['username']
+        return username
+
+    def post_message_to_channel(self, channel_name: str, message: str) -> None:
+        """
+        Post the message to the channel using the channel name.
+
+        Args:
+            channel_name: Name of the channel on mattermost.
+            message: Message to be posted on the channel.
```

Comment:
Nit: check out other docstrings in this repo. They're in `reStructuredText` format, while this is `Google Style`.